src/mds/FSMap.cc

```diff
@@ -868,6 +870,11 @@ void FSMap::sanity(bool pending) const
       ceph_assert(info.compat.writeable(fs.mds_map.compat));
     }
 
+    auto const& leader = fs.mds_map.get_quiesce_db_cluster_leader();
+    auto const& members = fs.mds_map.get_quiesce_db_cluster_members();
+    ceph_assert(leader == MDS_GID_NONE || members.contains(leader));
+    ceph_assert(std::ranges::all_of(members, [&infos = fs.mds_map.mds_info](auto m){return infos.contains(m);}));
```

Comment:
This last assert may backfire on you. If you ever change it so members may exist outside the MDSMap, then any monitor that decodes (update_from_paxos) the change (without being upgraded) will fail this sanity check.

However, I think the likelihood of that is low and we have in the past indicated that a monitor upgrade should disable the fsmap sanity checks for this type of problem (which we don't like doing).

Reply:
Well, I'll make sure to only update the logic once all monitors are on the new version.